src/components/lyne-link-button/lyne-link-button.tsx

```diff
@@ -0,0 +1,134 @@
+  Prop
+} from '@stencil/core';
+
+import getDocumentLang from '../../global/helpers/get-document-lang';
+import getDocumentWritingMode from '../../global/helpers/get-document-writing-mode';
+import i18n from '../../global/i18n';
+import { InterfaceLinkButtonAttributes } from './lyne-link-button.custom.d';
+
+/**
+ * @slot icon - Slot used to display the icon, if one is set
+ */
+
+@Component({
+  shadow: true,
+  styleUrls: {
+    default: 'styles/lyne-link-button.default.scss',
+    shared: 'styles/lyne-link-button.shared.scss'
+  },
+  tag: 'lyne-link-button'
+})
+
+export class LyneLinkButton {
+
+  /**
+   * If set to true, the browser will
+   * show the download dialog on click.
+   */
+  @Prop() public download?: boolean;
+
+  /** The href value you want to link to */
+  @Prop() public hrefValue!: string;
+
+  /**
+   * The icon name we want to use,
+   * choose from the small icon variants from
+   * the ui-icons category from here
+   * https://lyne.sbb.ch/tokens/icons/
+   */
+  @Prop() public icon?: string;
+
+  /**
+   * Decide whether the icon should get flipped
+   * horizontally if the document writing mode
+   * is changed from ltr to rtl or vice versa.
+   */
+  @Prop() public iconFlip?: boolean;
+
+  /**
+   * The icon can either be place before or after
+   * the text
+   */
+  @Prop() public iconPlacement: InterfaceLinkButtonAttributes['iconPlacement'] = 'start';
+
+  /**
+   * Pass in an id, if you need to identify
+   * the link element.
+   */
+  @Prop() public idValue?: string;
+
+  /** The link text we want to visually show */
+  @Prop() public text!: string;
+
+  /**
+   * Choose the link button style variant
+   */
+  @Prop() public variant: InterfaceLinkButtonAttributes['variant'] = 'primary';
+
+  public render(): JSX.Element {
+
+    const currentLanguage = getDocumentLang();
+    const currentWritingMode = getDocumentWritingMode();
+
+    let openInNewWindow = false;
+
+    if (!this.hrefValue.includes('sbb.ch')) {
+      openInNewWindow = true;
+    }
+
+    let iconPositionClass = '';
+
+    if (this.icon) {
+      iconPositionClass = ` link-button--icon-placement-${this.iconPlacement}`;
+    }
+
+    let iconFlipClass = '';
+
+    if (this.icon && this.iconFlip) {
+      iconFlipClass = ' link-button--icon-flip';
+    }
+
+    let addtitionalLinkAttributes = {};
+    let ariaLabel = this.text;
+
+    if (openInNewWindow) {
+      addtitionalLinkAttributes = {
+        rel: 'external noopener nofollow',
+        target: '_blank'
+      };
+      ariaLabel += `. ${i18n['modules'].link.targetOpensInNewWindow[currentLanguage]}`;
+    }
+
+    const variantClass = ` link-button--${this.variant}`;
+
+    return (
+      <a
+        aria-label={ariaLabel}
+        class={
+          `link-button
+          ${iconPositionClass}
+          ${iconFlipClass}
+          ${variantClass}`
+        }
+        download={this.download}
+        dir={currentWritingMode}
+        href={this.hrefValue}
+        id={this.idValue}
```

Comment:
i would handle this via `additionalLinkAttributes`, otherwise if no `id` is provided via attributes, we end up with an empty `id` attribute on the `a` tag